### DISCUSSION TITLE
NextJS - custom pageExtension makes no classname generation by both content and safelist

### DISCUSSION THREAD

**Discussion by spilist** [tailwindlabs/tailwindcss#8341] [Help] [answered]
**What version of Tailwind CSS are you using?**

- 3.0.24

**What build tool (or framework if it abstracts the build tool) are you using?**
- "next": "^12.1.6", (also tested with 12.1.7-canary.5 which contains https://github.com/vercel/next.js/pull/36747 but no luck)
- "react": "^18.1.0",
- "postcss": "^8.4.13",

**What version of Node.js are you using?**

- 16

**What browser are you using?**

- Latest Chrome

**What operating system are you using?**

- MacOS Monterey (12.1)

**Reproduction URL**

- https://stackblitz.com/edit/github-mcndf4
- I made it from https://tailwindcss.com/docs/guides/nextjs, and changed two things:
  - Add `  pageExtensions: ['page.tsx'],` on `next.config.js` (Refer to https://nextjs.org/docs/api-reference/next.config.js/custom-page-extensions#including-non-page-files-in-the-pages-directory)
  - Rename `pages/index.tsx` to `pages/index.page.tsx`
- Note that it uses `Webpack` instead of `swc` due to `.babelrc`.

**Describe your issue**

- When I add `pageExtensions` option to next.config.js, both `content` and `safelist` don't generate specified classnames. When I rolled back to default next config, they start working. 
- I suspected next's pageExtensions config somewhat changes how to resolve those pages, but still have no idea why `safelist` is not working either. I'm not sure where does this issue come from, but because of safelist I thought it also can be an issue of tailwindcss.

<img width="1595" alt="image" src="https://user-images.githubusercontent.com/1589430/168256237-40994edf-e7ed-45fc-8506-d411de0ffd1f.png">

**Comment by adamwathan** [comment]
Hey! Super weird one, it looks like Next.js is actually completely ignoring `globals.css` entirely when using custom page extensions:

https://stackblitz.com/edit/github-mcndf4-kjpkp1?file=styles/globals.css

<img width="1551" alt="image" src="https://user-images.githubusercontent.com/4323180/168292392-49ec4f82-a277-4367-90f1-64a09b63566e.png">

So I don't think this is a Tailwind CSS issue 🤔 Going to convert this to a discussion but hopefully can track down what's happening.

**Comment by adamwathan** [comment, marked as answer]
Solved, have to rename `_app.tsx` to `_app.page.tsx`:

https://stackblitz.com/edit/github-mcndf4-hitkzz?file=pages/_app.page.tsx

This is explained in the Next.js docs:

https://nextjs.org/docs/api-reference/next.config.js/custom-page-extensions

<img width="725" alt="image" src="https://user-images.githubusercontent.com/4323180/168386027-377fcb83-28fa-4a30-930f-6027a526ba1a.png">

**Comment by spilist** [comment]
Hi, thx for clarifying. Because my app still didn't work, so I tried to figure out what's different between stackblitz and my app.

Found out why: I specified "important" in tailwind config as 

```
  important: '__next',
```

which should have been 

```js
  important: '#__next',
```

and I got it working.

Anyway, thank you!
